cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -237,6 +237,29 @@ class ParquetScanTaskIterator {
   std::shared_ptr<parquet::arrow::FileReader> reader_;
 };
 
+Status ParquetFileFormat::IsSupported(const FileSource& source, bool* supported) const {
+  auto pool = default_memory_pool();
+
+  *supported = false;
+
+  std::unique_ptr<parquet::ParquetFileReader> reader;
+  auto status = OpenReader(source, pool, &reader);
```

Comment:
This seems like a fragile approach. I'd prefer `class ParquetInvalidFooterException : public ParquetException` which can be separately caught by `OpenReader`:
```suggestion
  bool invalid_footer = false;
  auto status = OpenReader(source, pool, &reader, &invalid_footer);
  if (invalid_footer) {
    // likely not a parquet file; ignore it
    return Status::OK();
  }
  RETURN_NOT_OK(status);
  *supported = true;
  return Status::OK();
```

Reply:
Alternatively, extract `parquet::HasValidFooter`